src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -198,21 +199,63 @@ public override IActorRef SystemActorOf<TActor>(string name = null)
         /// </summary>
         public override void Abort()
         {
-            Aborting = true;
+            Volatile.Write(ref Aborting, true);
             Terminate();
         }
 
         /// <summary>Starts this system</summary>
         public void Start()
+        {
+            using (var cts = new CancellationTokenSource(Settings.CreationTimeout))
+                StartAsync(cts.Token).GetAwaiter().GetResult();
+        }
+
+        /// <summary>Starts this system</summary>
+        public async Task StartAsync(CancellationToken cancellationToken = default)
         {
             try
             {
                 // Force TermInfoDriver to initialize in order to protect us from the issue seen in #2432
                 typeof(Console).GetProperty("BackgroundColor").GetValue(null); // HACK: Only needed for MONO
 
                 RegisterOnTermination(StopScheduler);
-                _provider.Init(this);
-                LoadExtensions();
+
+                //init provider
+                {
+                    _provider.Init(this);
+                    if (_provider is IInitializable init)
+                        await init.InitializeAsync(cancellationToken);
+                }
```

Comment:
Why would we call `Init` _and_ `InitializeAsync` ?

Reply:
`IInitializable` is a feature faced, it does not replace the old API